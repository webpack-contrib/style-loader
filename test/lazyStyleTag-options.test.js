/* eslint-env browser */

import {
  compile,
  getCompiler,
  getEntryByInjectType,
  getErrors,
  getWarnings,
  runInJsDom,
} from "./helpers/index";

describe("lazyStyleTag options", () => {
  it(`should pass "options" to "insert" function`, async () => {
    expect.assertions(3);

    const entry = getEntryByInjectType("options.js", "lazyStyleTag");
    const compiler = getCompiler(entry, {
      injectType: "lazyStyleTag",
      insert: (styleTag, options) => {
        options.insertInto.appendChild(styleTag);
      },
    });
    const stats = await compile(compiler);

    runInJsDom("main.bundle.js", compiler, stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot("DOM");
    });

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it(`should pass "options" to "insert" function and unuse`, async () => {
    expect.assertions(3);

    const entry = getEntryByInjectType("options-use-unuse.js", "lazyStyleTag");
    const compiler = getCompiler(entry, {
      injectType: "lazyStyleTag",
      insert: (styleTag, options) => {
        options.insertInto.appendChild(styleTag);
      },
    });
    const stats = await compile(compiler);

    runInJsDom("main.bundle.js", compiler, stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot("DOM");
    });

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it(`should pass "options" to "styleTagTransform" function`, async () => {
    expect.assertions(3);

    const entry = getEntryByInjectType("options.js", "lazyStyleTag");
    const compiler = getCompiler(entry, {
      injectType: "lazyStyleTag",
      styleTagTransform: (css, styleTag, options) => {
        // eslint-disable-next-line no-param-reassign
        styleTag.innerHTML = `${css}\n${options.additionalStyles}\n`;
      },
    });
    const stats = await compile(compiler);

    runInJsDom("main.bundle.js", compiler, stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot("DOM");
    });

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
