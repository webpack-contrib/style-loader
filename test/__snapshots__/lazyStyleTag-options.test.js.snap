// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazyStyleTag options should pass "options" to "insert" function and unuse: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`lazyStyleTag options should pass "options" to "insert" function and unuse: errors 1`] = `Array []`;

exports[`lazyStyleTag options should pass "options" to "insert" function and unuse: warnings 1`] = `Array []`;

exports[`lazyStyleTag options should pass "options" to "insert" function: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


<style>body {
  color: red;
}
</style></body></html>"
`;

exports[`lazyStyleTag options should pass "options" to "insert" function: errors 1`] = `Array []`;

exports[`lazyStyleTag options should pass "options" to "insert" function: warnings 1`] = `Array []`;

exports[`lazyStyleTag options should pass "options" to "styleTagTransform" function: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style>body {
  color: red;
}

.some-element {color: red;}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`lazyStyleTag options should pass "options" to "styleTagTransform" function: errors 1`] = `Array []`;

exports[`lazyStyleTag options should pass "options" to "styleTagTransform" function: warnings 1`] = `Array []`;
